### PR TITLE
Add Settings indicator for application-scoped settings

### LINKED ISSUE #152888: Introduce application model in configuration
In a non default profile window, configuration needs application model layer from which application scoped settings can be read/write.

- [x] Introduce application configuration model layer
- [x] Make sure application configuration is active and used only in non default profiles
- [x] read/write application scoped settings only from application layer and not from user layer
- [ ] Show application tab in settings editor for non default profiles

### PR DESCRIPTION
**Pull request by rzhao271**
This PR fixes #152888

## Changes in this PR

- Adds indicators to application-scoped settings with custom hovers. These indicators currently always show up, even for the default profile.
- Adds aria text for those indicators.
- Changes the link for "Edit in settings.json" to go to the right file when a non-default profile is active.
- Changes the action button to have a dropdown when in a non-default profile.
- Fades out profile-specific settings when editing the default profile settings JSON file.
- Fades out application-scoped settings when editing the profile settings JSON file.

## Screenshots

Example showing a faded out profile setting in the default profile JSON file while the user has a non-default profile active
The new dropdown that displays when switching to the JSON file from the User tab in the GUI Settings editor, showing the options "Open Default Profile Settings (JSON)" and "Open Profile Settings (JSON)"
Example showing a faded out application-scoped setting in the profile JSON file and a message saying that they must set the setting in the default profile JSON file
The new "Applies to all profiles" indicator on the telemetry.telemetryLevel setting in the GUI Settings editor

### CODE REVIEW

**Comment by rzhao271** [comment]
I would rather add in the line fading for the profile-specific settings on the application-scoped settings JSON file later.
Marking as ready for review.

File: src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts [hunk 1]
```diff
@@ -267,31 +268,89 @@ class PreferencesActionsContribution extends Disposable implements IWorkbenchCon
...
+							return editorPane.switchToSettingsFile();
+						}
+						return null;
+					}
+				}));
+				openJsonFromUserSettingsDisposableStore.add(registerAction2(class extends Action2 {
+					constructor() {
+						super({
+							id: SETTINGS_EDITOR_COMMAND_SWITCH_TO_APPLICATION_JSON,
+							title: { value: nls.localize('openApplicationSettings', "Open Non-Profile Specific Settings (JSON)"), original: 'Open Non-Profile Specific Settings (JSON)' },
```
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts:298]: Any ideas for how to make this label shorter? One suggestion was "Open Global Settings". Another one I have is "Open settings that apply to all profiles", though that's not necessarily true; it's more like "Open Default Profile Settings (JSON)".

**Comment by rzhao271** [comment]
Added in some screenshots.

**Comment by roblourens** [comment]
I haven't been part of all the discussions but all the language here is a little confusing. What is a non-default profile vs a default profile? What is "default profile user settings", that implies that a profile has user settings vs workspace settings?

And you demoed this in standup but I didn't really follow, but when you set the telemetry level, I think it was setting it in multiple JSON files, is that right?

**Comment by rzhao271** [comment]
With the Settings profile feature enabled, the first profile the user gets is the default profile. The user can then create new profiles. All application-scoped settings are stored to the default profile settings JSON file. When setting an application-scoped setting in the GUI editor, the application-scoped setting is always stored only to the default profile settings JSON file. Application-scoped setting values are read from the default profile settings JSON file.
When the Settings profile feature is disabled, application-scoped settings and other user settings are all saved into the same file. That file becomes the default profile settings JSON file when the feature is enabled.

**Comment by roblourens** [comment]
Thanks. I think it's confusing to me that the "default profile" is like a layer that is active at the same time as a non-default profile. I think there should only be one profile active at a time. It would make sense if there was a default profile, but I could also create other profiles, and switch from the default profile to the other profile. But here it sounds like some settings go to the default profile, some go to the other, and the other gets overlayed on another. If that's the case, why not just call the place where application settings go the "user settings", and a profile gets overlayed on it?

Some other possibilities could be calling it the "base profile" or "user settings profile" or something like that. But still it feels weird to me to say that I have two profiles. The language "applies to all profiles" makes it sound like there could be even more.

**Comment by rzhao271** [comment]
Without Settings profiles, the application-scoped settings and non-application-scoped user settings are saved into the same file.
For backwards compatibility, when Settings profiles are enabled, that file becomes the default profile `settings.json` file. When the user has a non-default profile open, only the application-scoped settings from that file are used. The rest of the user settings come from another json file specific to the profile. When the user has the default profile open, all user settings are saved to that one file, so we don't need the dropdown anymore asking which settings json file the user wants to open.
For now, the user can have at most one profile open at a time in a window. There could be more profiles, but there's no nesting for now other than the fact that application-scoped settings are saved in a separate file from the profile JSON file if the user has a non-default profile open.

**Comment by rzhao271** [comment]
I'll be renaming some of the labels according to the feedback above. For example, Default Profile -> User.

**Comment by roblourens** [comment]
I get the model, I'm just still thinking through what the user will think. Btw, I noticed that the hint appears even when profiles are not enabled:

image

The blue bar doesn't appear when I enable this setting

image

At first I was thinking it's just confusing that we are using one profile file as a store for application-scope settings, but I might be coming around to it. Did you consider migrating application-scope settings to another file entirely which would not be a profile, but just a place to store user settings which can't go into a profile?

**Comment by rzhao271** [comment]
## Main goal

We want to indicate application-scoped settings in the Settings editor, but we want to talk about them in terms of profiles, so that the Settings profiles concept only introduces the word "profiles". We want users to only have to know what a profile is, rather than also have to learn what an application-scoped setting is, unless they're an extension dev.

## Newer screenshots

An application-scoped setting in the Settings editor with the indicator "Not profile-specific" and custom hover text "The setting is not specific to the current profile, and will retain its value when switching profiles."
Dropdown with options "Open Profile Settings (JSON)" and "Open User Settings (JSON)"
ARIA label text "Setting value retained when switching profiles" for an application-scoped setting

**Comment by roblourens** [comment]
I'm not sure - I think I like "Applies to all profiles better".

Also I thought that to reduce noise, maybe you could only show that hint when the setting is modified. I'm not sure it really matters that much ahead of time.

**Review by roblourens — APPROVED** [review]

**Pull request merged**